getting_started/Performance_Metrics.md

```diff
@@ -0,0 +1,75 @@
+We divide this by the measured step time (measured via python `time.time()`), to get the Model Flops per second that we print each step [`per_device_tflops_seconds`](https://github.com/AI-Hypercomputer/maxtext/blob/e969faabbb571285a51545530f34d8f0a9f237e9/MaxText/metric_logger.py#L193-L194). You can get the MFU by diving this number by the peak tflops of the hardware (e,g. $918e^{12}$ FLOPS/s for trillium)
+
+ $$ MFU = \frac{\text{model flops/s}}{\text{peak hardware flops/s}} = \frac{\text{theoretical model flops}}{\text{measured step time} \times \text{peak hardware flops/s}}$$
+
+This is also equivalent to the ratio of the theoretically optimal step time and measured step time:
+
+$$\text{theoretically optimal step time} = \frac{\text{theoretical model flops}}{\text{peak hardware flops/s }} $$
+
+$$\frac{\text{theoretically optimal step}}{\text{measured step time}} = \frac{\text{theoretical model flops}}{\text{measured step time} \times \text{peak hardware flops/s }} = MFU$$
+
+
+### Casual Attention
+Due to causality only half of the (query, key) pairs need to be computed, those with query_idx >= key_idx. This accounts for the fact only prior tokens can be used to predict future ones. Prior to https://github.com/AI-Hypercomputer/maxtext/pull/1988 MaxText did not account for sparsity for theoretical flops, and used
+
+Attention Flops ~= 4 * sequence^2 * batch * heads * head_dim
+
+
+When accounting for casual masking, this should be halved
+
+Attention Flops ~= 2 * sequence^2 * batch * heads * head_dim
+
+Which maxtext now uses since this [PR/1988](https://github.com/AI-Hypercomputer/maxtext/pull/1988)
+
+Note that
+
+$$ \text{Total Flops} =  \text{Attention (quadratic in sequence) + Non-attention  (linear)}$$ 
+
+Thus the distinction between casual vs non casual flops is particularly important for long sequence when the attention flops dominate / are a significant fraction of the total flops. For 8k sequence length, the attention flops are generally around 10% of total flops (depending on exact model dims), whereas for 128k seq, the attention flops may be around 90%. Note however the attention flops also vary by attention type, e.g. sliding window flops are not quadratic in sequence.
+
+
+
+### Why MFU
+MFU is a very useful metric to understand your systems performance, but like step time or tokens/s, there are pros and cons of summarizing the system’s performance to a single number.
+
+**Pros**
+* Clearly shows room left to improve, how much more the hardware is capable of. (e.g. 25% MFU means its possible to get 4x more performance and 4x smaller step times)
+* Generalizable across hardwares, model, configs (e.g. batch sizes)
+
+**Cons**
+* Have to be cognizant of exact formula used to calculate model flops (e.g. was causality taken into account?)
```

Comment:
Probably rephrase to hard to compare across platforms/repositories due to formula differences?